### DISCUSSION TITLE
Broken Images

### DISCUSSION THREAD

**Discussion by micogg** [blakeblackshear/frigate#2632] [Ask A Question] [answered]
Hi,

I’m having an issue with the streams breaking.  The break constantly changes.  Sometimes I cannot see anything in the image and sometimes it’s crystal clear.  I have attached a picture for reference.

To be honest, I don’t know where to start troubleshooting.  I assume someone has run into this before and can maybe point me in a direction to look.

Thank you in advance.

![F504AE1A-2C19-4551-A1BB-3D3732112344](https://user-images.githubusercontent.com/33884470/149232141-5f0dcbfd-e42b-44c6-a5c3-76919a90e54d.jpeg)

**Comment by blakeblackshear** [comment, marked as answer]
This is usually because your camera is using udp or because you are using Reolink cameras and not using the recommended arguments from the documentation. Please post your config file and ffprobe output from your camera if you want help.

**Reply by micogg** [reply]
Thanks for the follow up.  I did not see the camera specific documentation.  As you called it, they are Reolink cameras.  
I actually found another [thread](https://github.com/blakeblackshear/frigate/issues/1713) that helped me get the http stream working, but I'm now having issues with the FFMPEG process crashing.  When I reboot Frigate, the camera that send the broken frame seems to periodically change. 

Errors:

022-01-13T16:18:10.398535197Z [2022-01-13 16:18:10] ffmpeg.front left.detect       ERROR   : [rtmp @ 0x562280e32d00] Server error: Already publishing
2022-01-13T16:18:10.398566616Z [2022-01-13 16:18:10] ffmpeg.front left.detect       ERROR   : rtmp://127.0.0.1/live/front left: Operation not permitted
2022-01-13T16:18:12.561276656Z [2022-01-13 16:18:12] frigate.video                  INFO    : front left: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
2022-01-13T16:18:12.561394096Z [2022-01-13 16:18:12] frigate.video                  INFO    : front left: ffmpeg process is not running. exiting capture thread...
2022-01-13T16:18:20.405944010Z [2022-01-13 16:18:20] watchdog.front left            ERROR   : FFMPEG process crashed unexpectedly for front left.
2022-01-13T16:18:20.406117755Z [2022-01-13 16:18:20] watchdog.front left            ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2022-01-13T16:18:20.406253590Z [2022-01-13 16:18:20] watchdog.front left            ERROR   : You may have invalid args defined for this camera.
2022-01-13T16:18:20.406834268Z [2022-01-13 16:18:20] ffmpeg.front left.detect       ERROR   : [rtmp @ 0x559c6215d700] Detected librtmp style URL parameters, these aren't supported by the libavformat internal RTMP

Config:

objects:
  track:
    - person
record:
  enabled: True
  retain_days: 0
  events:
    objects:
      - person
    retain:
      default: 10
      objects:
        person: 15
snapshots:
  enabled: True
  timestamp: True
  retain:
    default: 10
    objects:
      person: 15
ffmpeg:
  hwaccel_args:
    - -c:v
    - h264_cuvid
  input_args:
    - -avoid_negative_ts
    - make_zero
    - -fflags
    - nobuffer+genpts+discardcorrupt
    - -flags
    - low_delay
    - -strict
    - experimental
    - -analyzeduration
    - 1000M
    - -probesize
    - 1000M
    - -rw_timeout
    - "5000000"
  output_args:
    record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
cameras:
  back:
    ffmpeg:
      inputs:
        - path: http://192.168.1.154/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=username&password=password
          roles:
            - record
            - rtmp
  front left:
    ffmpeg:
      inputs:
        - path: http://192.168.1.175/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=username&password=password
          roles:
            - record
            - rtmp
    motion:
      mask:
        - 611,165,1280,215,1280,0,0,0,0,187
        - 354,160,0,167,0,438
  front right:
    ffmpeg:
      inputs:
        - path: http://192.168.1.153/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=username&password=password
          roles:
            - record
            - rtmp
    detect:
      width: 640
      height: 480
      fps: 7
      enabled: True
    motion:
      mask:
        - 652,148,1280,179,1280,0,0,0,0,181
        - 1083,162,1280,178,1280,284
    zones:
      right_yard:
        coordinates: 0,720,1280,720,1280,267,1236,247,886,235,573,237,0,252

**Reply by micogg** [reply]
Thanks Blake! Everything is working now.  I found in another thread you assisted in a mention of disabling rtmp.  Once I did that, it resolved all the errors.

The "front right" camera is in a different aspect ratio than the other two, but I'm sure it's something simple so I should get it resolved. 

You've killed it with this program, solid job coding and with the support!

**Reply by gulo101** [reply]
Hi @micogg  , what exact model of camera do you have and are the settings in your post above what resolved your issue? Thanks

**Reply by micogg** [reply]
My current config with the stable feeds are in issue #2647 .  That issue was unrelated to the tearing but instead because I can't get the correct aspect ratio.  The original tearing issue was resolved by changing from the rtsp to http feed but the input/output args were required for the http feed to work.  I am not sure which arguments were needed, I found them all in a different discussion and just copied all of them.  There is no more tearing and the feed is notably smoother both live and on recordings.

The only current issue I'm experiencing with this config is that all the feeds are 4:3 instead of 16:9.  The aspect ratio changed after switching to http and adding all the input/output args.  I'm not sure if it's related or I unintentionally changed something else.

The cameras are Reolink RLC-410-5MP

Lastly, there is actually documentation for the Reolink cameras on the Frigate website that might help.  [https://docs.frigate.video/configuration/camera_specific/](url)

**Comment by gregewing** [comment]
I know this is closed, but I wanted to comment on it that with the cameras I have, i'm unable to force them to send video on TCP.  The default behaviour is UDP.  The cameras have a dialog to force TCP, but it has no effect.  ( Hikvision grey imports ).

I eventually resolved the issue by fording frigate to talk TCP to teh cameras with this additional input_arg for each camera that is affected:

```  garage:
    enabled: true
    ffmpeg:
      input_args:
        - -c:v
        - h264_cuvid
        - -rtsp_transport          #  <-- added in these two lines
        - tcp                      #  <-- added in these two lines
      inputs:
        - path: rtsp://user:pw@192.168.1.65:554/h264Preview_01_sub
          roles:
            - detect
            - record
```
